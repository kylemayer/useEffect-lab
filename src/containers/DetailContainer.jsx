import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { fetchVillagers } from '../services/animalCrossingApi';

const DetailContainer = () => {
  const { id } = useParams();
  const [loading, setLoading] = useState(true);
  const [villager, setVillager] = useState({});

  useEffect(() => {
    fetchVillagers(id)
      .then((villager) => setVillager(villager))
      .finally(() => setLoading(false));
  }, [id]);

  if(loading) return <h1>Loading villager...</h1>;

  return (
    <article>
      <img src={villager.image} alt={villager.name}/>
      <p>{villager.name} ({villager.gender})</p>
      <p>{villager.phrase}</p>
    </article>
  );
};

export default DetailContainer;
