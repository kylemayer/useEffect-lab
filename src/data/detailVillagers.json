{
    "_id": "5f5fb4bbbfd05c2aed82e472",
    "url": "https://animalcrossing.fandom.com/wiki/Astrid",
    "name": "Astrid",
    "japaneseName": "キッズ Kizzu",
    "image": "https://vignette.wikia.nocookie.net/animalcrossing/images/d/d5/Astrid_NH.png/revision/latest?cb=20200802154954",
    "quote": "An apple doesn't fall far from the tree.",
    "gender": "female",
    "personality": "snooty",
    "species": "kangaroo",
    "birthday": "September 8th (Virgo)",
    "phrase": "my pet",
    "skill": "Karaoke",
    "goal": "Singer",
    "coffee": {
        "roast": "Blue Mountain",
        "milk": "Just a little bit of milk",
        "sugar": "One spoonful of sugar"
    },
    "song": {
        "name": "K.K. Gumbo",
        "link": "https://animalcrossing.fandom.com/wiki/K.K._Gumbo"
    },
    "style": "Rock 'n' roll",
    "__v": 0
}