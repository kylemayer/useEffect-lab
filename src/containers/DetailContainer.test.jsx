/* eslint-disable max-len */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter } from 'react-router';
import DetailContainer from './DetailContainer';
import detailVillagers from '../data/detailVillagers.json';

const server = setupServer(
  rest.get('https://ac-vill.herokuapp.com/villagers/:id', (req, res, ctx) => {
    return res(ctx.json(detailVillagers));
  })
);

describe('DetailContainer', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('display a single villager', async () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/5f5fb4bbbfd05c2aed82e472']} keyLength={25}>
        <DetailContainer />
      </MemoryRouter>
    );

    screen.getByText('Loading villager...');

    expect(container).toMatchSnapshot();
  });
});
