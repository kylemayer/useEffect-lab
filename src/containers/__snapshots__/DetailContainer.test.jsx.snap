// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailContainer display a single villager 1`] = `
<div>
  <h1>
    Loading villager...
  </h1>
</div>
`;
